### DISCUSSION TITLE
Status Update Lazy Loading

### DISCUSSION THREAD

**Discussion by kleelockerby** [mrpmorris/Fluxor#372] [General]
Peter,
I read all of your comments regarding the issues surrounding Lazy Loading in 
gitter (https://gitter.im/mrpmorris/fluxor?at=5fd155a85a63c561051e5391). 

I noticed it was dated August 2020. Have there been any advancements in this area since then? 

Also, we are using ver 4.1, will upgrading to 5.6 make a difference.

We are an enterprise level financial application servicing Fortune 100 Companies that is very resource intensive. We are well entrenched with Fluxor and need to go down the Lazy Loading Path as well.
Thanks very much!

**Comment by mrpmorris** [comment]
The link you gave isn't to a lazy loading discussion.

I would create all the store in one dll that is available on startup, and lazy load the UI parts.

It's doable, but you would have to new everything up in the lazy module yourself.

**Reply by kleelockerby** [reply]
That's strange, it was for me, but I've found gitter.im strange to work with, here's what I was referring to:
![Screenshot_1](https://user-images.githubusercontent.com/20687752/204921222-da99b766-4697-4b46-949d-616a310a0f5d.png)

I apologize for the convenience.
Thank you.

**Comment by mrpmorris** [comment]
I suppose it would be possible to auto create the registration code using a roslyn generator, which could then make it easy to dynamically load a dll and register it.

You'd have to ensure all dependencies of your effects are registered during app start though.
